WireDomain/Sources/WireDomain/Event Processing/UserEventProcessor/UserConnectionEventProcessor.swift

```diff
@@ -32,9 +33,248 @@ protocol UserConnectionEventProcessorProtocol {
 
 struct UserConnectionEventProcessor: UserConnectionEventProcessorProtocol {
 
-    func processEvent(_: UserConnectionEvent) async throws {
-        // TODO: [WPB-10191]
-        assertionFailure("not implemented yet")
+    enum UserConnectionEventError: Error {
+
+        case failedToFetchConversation
+        case failedToEstablishGroup(Error)
+        case failedToActivateConversation
+
+    }
+
+    let context: NSManagedObjectContext
+    let connectionsRepository: any ConnectionsRepositoryProtocol
+    let userRepository: any UserRepositoryProtocol
+    let conversationsRepository: any ConversationRepositoryProtocol
+    let mlsService: any MLSServiceInterface
+
+    func processEvent(_ event: UserConnectionEvent) async throws {
+        let connection = event.connection
+
+        try await connectionsRepository.updateConnection(connection)
+
+        if connection.status == .accepted {
+            try await resolveOneOnOneConversation(for: connection.receiverQualifiedID)
```

Comment:
I wanted to encapsulate all that resolving 1:1 conversation logic into a `ResolveOneOnOneConversationUseCase` but I wasn't sure whether it made sense. IMO, it makes sense to create use cases and expose them publicly if it's going to be used by WireDomain clients.

Reply:
Following up on my last comment, once this [ticket](https://wearezeta.atlassian.net/browse/WPB-10727) is merged, we'll be able to reuse `OneOnOneResolverUseCase` here to avoid code duplication.